### PR TITLE
MINOR: Clean up core metrics, migration, network, raft, security, serializer, tools, utils, and zookeeper modules

### PR DESCRIPTION
**Pull request by jlprat**
Second follow up from https://github.com/apache/kafka/pull/15252

This PR cleans up: metrics, migration, network, raft, security, serializer, tools, utils, and zookeeper package classes

- Mark methods and fields private where possible
- Annotate public methods and fields
- Remove unused classes and methods
- Make sure Arrays are not printed with .toString
- Optimize minor warnings

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala [hunk 1]
```diff
@@ -37,6 +37,7 @@ import org.apache.kafka.common.utils.Time
 import org.apache.kafka.common.utils.Utils
 import org.apache.kafka.server.util.{CommandDefaultOptions, CommandLineUtils}
 
+import java.util
```
Comment (mimaison) [inline core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: I've got a PR to move this to the tools module: https://github.com/apache/kafka/pull/15274 so I'm not sure it's worth updating it.

**Review by mimaison — COMMENTED** [review]
Thanks for the PR.

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala (hunk 1 above)
Comment (jlprat) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: I can review your PR first, and then rebase this one, so this class is gone :)

Comment (mimaison) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: Works for me 👍

File: core/src/main/scala/kafka/network/RequestChannel.scala [hunk 2]
```diff
@@ -92,18 +92,18 @@ object RequestChannel extends Logging {
...
+    @volatile var responseDequeueTimeNanos: Long = -1L
+    @volatile var messageConversionsTimeNanos: Long = 0L
+    @volatile var apiThrottleTimeMs: Long = 0L
+    @volatile var temporaryMemoryBytes: Long = 0L
     @volatile var recordNetworkThreadTimeCallback: Option[Long => Unit] = None
     @volatile var callbackRequestDequeueTimeNanos: Option[Long] = None
     @volatile var callbackRequestCompleteTimeNanos: Option[Long] = None
 
-    val session = Session(context.principal, context.clientAddress)
+    val session: Session = Session(context.principal, context.clientAddress)
```
Comment (OmniaGM) [inline core/src/main/scala/kafka/network/RequestChannel.scala]: I see sometimes we add the type for `val` and `var` definition and sometimes we don't am just curious what is the pattern we are opting in for here?

Comment (jlprat) [reply core/src/main/scala/kafka/network/RequestChannel.scala]: I might have missed some spots where this should be applied but the Scala idiomatic approach to this is:
- public API should be explicitly typed: this applies to `val`s, `var`s and `def`s for example. Like the case you comment on.
- private definitions or variables don't need to be typed. Thanks to Scala type inference many `def`s, `var`s and `val`s can omit their explicit type making it easy to refactor things. Please note the "don't need to", you can still explicitly type those.

Let me know if this answers your question.

Comment (jlprat) [reply core/src/main/scala/kafka/network/RequestChannel.scala]: The changes made in this PR reflect the current (as per 2.12 and 2.13) idiomatic way to write Scala. For example:
Boolean parameters should be named, idempotent methods that are parameterless can be omit the parenthesis, side-effecting parameterless methods should include the parenthesis, and so on

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala (hunk 1 above)
Comment (mimaison) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: #15274 is merged now

Comment (jlprat) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: Rebased against trunk. Feel free to review whenever you have time @mimaison. Thanks!

**Review by mimaison — APPROVED** [review]
LGTM, thanks

**Comment by jlprat** [comment]
Checked that all tests pass locally and are flaky tests.

**Pull request merged**